pages/tic-tac-toe.tsx

```diff
@@ -0,0 +1,135 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import styled from "styled-components";
+import { CenteredMain } from "../components/Centered";
+import { checkAnswer } from "../services/checkAnswer";
+
+const LOCK_LETTER = 'T'
+const Container = styled.div`
+    display: grid;
+    grid-template-columns: 200px 200px 200px;
+    grid-template-rows: 200px 200px 200px;
+`;
+
+const Cell = styled.div`
+    border: 1px solid black;
+    text-align: center;
+    vertical-align: middle;
+    font-size: 12em;
+    user-select: none;
+    cursor: pointer;
+`
+
+const XCell = styled(Cell)`color: blue`;
+const OCell = styled(Cell)`color: Purple`;
+
+const ClickedXCell = () => {
+    return <XCell>X</XCell>
+}
+
+const ClickedOCell = () => {
+    return <OCell>O</OCell>
+}
+
+const winningPatterns = [
+  [0,1,2],
+  [0,3,6],
+  [1,4,7],
+  [2,5,8],
+  [0,4,8],
+  [2,4,6],
+  [6,7,8],
+  [3,4,5],
+]
+
+export default function T() {
+  const router = useRouter();
+  const cells = Array.from({ length: 9 }, (_, i) => i);
+
+  const [remainingChoices, setRemainingChoices] = useState(cells);
+
+  const [oIndexes, setOIndexes] = useState<number[]>([]);
+  const [xIndexes, setXIndexes] = useState<number[]>([]);
+  const [yourTurn, setYourTurn] = useState(true)
+
+  const findMoveForRobot = (yourChoices: number[]): number | undefined => {
+    for (let i = 0; i < winningPatterns.length; i++) {
+        const commonNumbers = winningPatterns[i].filter(num => yourChoices.includes(num))
+        let choice;
+        if (commonNumbers.length === 2) {
+            choice = winningPatterns[i].find((num) => !yourChoices.includes(num))
+        }
+
+        if (choice && !oIndexes.includes(choice)) {
+            return choice
+        } else {
+            continue
+        }
+    }
+  }
+
+  const checkWinner = (choices: number[]) => {
+    return winningPatterns.some(pattern => {
+      return pattern.filter(num => choices.includes(num)).length === 3
+    })
+  }
+
+  const switchTurn = (index: number) => {
+      if (!yourTurn) {
+          return;
+      }
+      const allMyChoices = [index, ...xIndexes]
+      setXIndexes(allMyChoices)
+      setYourTurn(false)
+
+      const youWon = checkWinner(allMyChoices)
+
+      if (youWon) {
+        checkAnswer(LOCK_LETTER, true, router)
+        return;
+      }
+
+      const choicesMinusMine = remainingChoices.filter(item => item !== index)
+      if (choicesMinusMine.length === 0) {
+        checkAnswer(LOCK_LETTER, false, router)
+        router.reload();
+      }
+
+      setRemainingChoices(choicesMinusMine)
+
+      setTimeout(() => {
```

Comment:
The robot's turn is done after a 2 second delay, just to give the feeling that the robot is "thinking"